src/main.cpp

```diff
@@ -35,6 +35,7 @@
 
 ADDON::CHelper_libXBMC_addon *xbmc = 0;
 std::uint16_t kodiDisplayWidth(0), kodiDisplayHeight(0);
+std::string pipe;
```

Comment:
Sure thing. I've slowly managed to figure it out - I've got it to take the info from the listitem property, store in the session, pass the info to the adaptivetree_ instance so it can be passed to the download function in main. 

Working now on how to parse the curl options string